### ISSUE TITLE
Dependency Updates

### ISSUE THREAD

**Issue by ParaskP7** [wordpress-mobile/WordPress-Android#17551]
This is the parent issue for the `Dependency Updates for WPAndroid` project.

Platform Request: N/A
Project Thread: `paaHJt-49c-p2`

The `Dependency Updates for WPAndroid` project is about updating as many of WPAndroid's dependencies as possible, but within a timeboxed period of time (2 sprints).

-----

### Tasks

- Test
    - [x] #17554 @ParaskP7
    - [x] #17555 @ParaskP7
    - [x] #17556 @oguzkocer
        - [ ] TODO (⚙️) on `Espresso`
    - [x] #17557 @oguzkocer
    - [x] #17558 @ParaskP7
        - [ ] TODO (⚙️) on `WireMock`
- Main
    - [x] #17559 @oguzkocer @ParaskP7
    - [x] #17560 @ParaskP7
    - [x] #17561 @ParaskP7 @pachlava
        - [x] Blocked By (🚫) #17560
    - [x] #17562 @ParaskP7
    - [x] #17563 @ParaskP7 @pachlava
        - [x] Blocked By (🚫) #17790
    - [x] #17564 @ParaskP7
    - [x] #17565 @ParaskP7
    - [x] #17566 @ParaskP7 
        - [x] ~~Pending On (⚠️) `ExoPlayer` & `Gutenberg Mobile`~~ (📕)
    - [x] #17567 @ParaskP7
        - [x] `Firebase` Blocked By (🚫) #17563
    - [x] #17935 @ParaskP7
        - [x] Blocked By (🚫) #17563
    - [x] #17798 @ParaskP7
        - [x] Blocked By (🚫) #17563
    - [x] #17568 @ParaskP7
    - [x] #17569 @ParaskP7
    - [ ] #17570 @ParaskP7
        - [x] `KotlinPoet` Blocked By (🚫) #17563
    - [x] #17571 @ParaskP7

-----

### Stretch Goals:

- [ ] #17903
- [ ] Dependency Updates - Libs
    - [ ] Aztec
        - [x] Kotlin (currently `1.4.20`) -> [#1015](https://github.com/wordpress-mobile/AztecEditor-Android/pull/1015)
        - [x] Compile Warnings As Errors -> [#1016](https://github.com/wordpress-mobile/AztecEditor-Android/pull/1016)
        - [x] Compile Sdk (currently `28`) -> [Comment](https://github.com/wordpress-mobile/AztecEditor-Android/issues/1003#issuecomment-1403624740) -> [#1037](https://github.com/wordpress-mobile/AztecEditor-Android/pull/1037) -> [#1040](https://github.com/wordpress-mobile/AztecEditor-Android/pull/1040)
        - [ ] AndroidX Core
            - [ ] Legacy Support v4 (removal)
            - [ ] Appcompat (currently `1.0.0`)
    - [ ] Stories
        - [x] Kotlin (currently `1.4.20`)
        - [x] Compile Warnings As Errors
        - [ ] Android Core
            - [ ] Lifecycle (currently `2.2.0`)
            - [ ] Core (currently `1.2.0`)
            - [ ] Appcompat (currently `1.0.2`)
        - [ ] Android Other
            - [ ] Emoji (currently `1.0.0`)
- [x] #17577
- [ ] #19552

-----

### Issues Created:

- [x] #17962

**Comment by ParaskP7** [comment]
All tasks are now complete, excluding the stretch goals and #17570 (ish, see [comment](https://github.com/wordpress-mobile/WordPress-Android/issues/17570#issuecomment-1798515638)), and as such this parent issue can now be closed! 🎉

PS: From now on, all the stretch goals related tasks will be done in isolation to this parent task.
